src/main/java/be/seeseemelk/mockbukkit/entity/PlayerMock.java

```diff
@@ -177,6 +185,52 @@ public PlayerMock(ServerMock server, String name, UUID uuid)
 		closeInventory();
 	}
 
+	/**
+	 * Simulates a disconnection from the server.
+	 *
+	 * @return True if the player was disconnected, false if they were already offline.
+	 */
+	public boolean disconnect()
+	{
+		if (!online)
+		{
+			return false;
+		}
+		this.online = false;
+		this.lastPlayed = System.currentTimeMillis();
+
+		PlayerQuitEvent playerQuitEvent = new PlayerQuitEvent(this, String.format(QUIT_MESSAGE, this.getDisplayName()));
+		Bukkit.getPluginManager().callEvent(playerQuitEvent);
+
+		this.server.getPlayerList().disconnectPlayer(this);
+
+		return true;
+	}
+
+	/**
+	 * Simulates a connection to the server.
+	 *
+	 * @return True if the player was connected, false if they were already online.
+	 */
+	public boolean connect()
```

Comment:
I think it would make more sense to call this method `reconnect` as it can only have a meaningful effect if the player was already connected, then disconnected.

Another reason as to why `connect()` might not be the best choice is the possibility of people wrongfully assuming that this method needs to be called in order for a player to be connected, then  wondering why it throws an `IllegalStateException`. The exception context "Player wasn't online before" also pretty much implies that this method is about **re**-connecting the player.

```suggestion
	public boolean reconnect()
```